synapse/storage/databases/main/cache.py

```diff
@@ -223,7 +223,7 @@ def _invalidate_caches_for_event(
         # process triggering the invalidation is responsible for clearing any external
         # cached objects.
         self._invalidate_local_get_event_cache(event_id)
-        self.have_seen_event.invalidate((room_id, event_id))
+        self.have_seen_event.invalidate(((room_id, event_id),))
```

Comment:
It might be worth additionally adding a test where we try this in worker mode because even though `_invalidate_caches_for_event` was called via events over replication, the `have_seen_event` key problem here meant it still didn't work.